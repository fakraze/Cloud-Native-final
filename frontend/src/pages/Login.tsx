import React, { useState } from 'react';
import { Navigate } from 'react-router-dom';
import { Eye, EyeOff, Mail, Lock } from 'lucide-react';
import { useLogin } from '../hooks/useAuth';
import { useAuthStore } from '../store/authStore';

const Login: React.FC = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [showPassword, setShowPassword] = useState(false);
  const { isAuthenticated } = useAuthStore();
  const loginMutation = useLogin();

  if (isAuthenticated) {
    return <Navigate to="/restaurant" replace />;
  }

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    loginMutation.mutate({ email, password });
  };

  return (
    <div className="min-h-screen bg-gray-50 flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8">
      <div className="max-w-md w-full space-y-8">
        <div>
          <div className="mx-auto h-12 w-12 bg-blue-600 rounded-lg flex items-center justify-center">
            <Mail className="h-6 w-6 text-white" />
          </div>          <h2 className="mt-6 text-center text-3xl font-extrabold text-gray-900">
            Sign in to your account
          </h2>
          <p className="mt-2 text-center text-sm text-gray-600">
            Restaurant Ordering System
          </p>
          {import.meta.env.VITE_USE_MOCK_API === 'true' && (
            <div className="mt-4 p-3 bg-blue-50 border border-blue-200 rounded-lg">
              <p className="text-xs text-blue-800 font-medium mb-2">ðŸ§ª Demo Mode - Test Credentials:</p>
              <div className="text-xs text-blue-700 space-y-1">
                <div><strong>Employee:</strong> employee@test.com / password123</div>
                <div><strong>Admin:</strong> admin@test.com / password123</div>
              </div>
            </div>
          )}
        </div>

        <form className="mt-8 space-y-6" onSubmit={handleSubmit}>
          <div className="space-y-4">
            <div>
              <label htmlFor="email" className="block text-sm font-medium text-gray-700">
                Email address
              </label>
              <div className="mt-1 relative">
                <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
                  <Mail className="h-5 w-5 text-gray-400" />
                </div>
                <input
                  id="email"
                  name="email"
                  type="email"
                  autoComplete="email"
                  required
                  value={email}
                  onChange={(e) => setEmail(e.target.value)}
                  className="input-field pl-10"
                  placeholder="Enter your email"
                />
              </div>
            </div>

            <div>
              <label htmlFor="password" className="block text-sm font-medium text-gray-700">
                Password
              </label>
              <div className="mt-1 relative">
                <div className="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
                  <Lock className="h-5 w-5 text-gray-400" />
                </div>
                <input
                  id="password"
                  name="password"
                  type={showPassword ? 'text' : 'password'}
                  autoComplete="current-password"
                  required
                  value={password}
                  onChange={(e) => setPassword(e.target.value)}
                  className="input-field pl-10 pr-10"
                  placeholder="Enter your password"
                />
                <button
                  type="button"
                  className="absolute inset-y-0 right-0 pr-3 flex items-center"
                  onClick={() => setShowPassword(!showPassword)}
                >
                  {showPassword ? (
                    <EyeOff className="h-5 w-5 text-gray-400" />
                  ) : (
                    <Eye className="h-5 w-5 text-gray-400" />
                  )}
                </button>
              </div>
            </div>
          </div>

          {loginMutation.error && (
            <div className="bg-red-50 border border-red-200 rounded-lg p-3">
              <p className="text-sm text-red-600">
                Login failed. Please check your credentials and try again.
              </p>
            </div>
          )}

          <div>
            <button
              type="submit"
              disabled={loginMutation.isPending}
              className={`w-full btn-primary ${
                loginMutation.isPending ? 'opacity-50 cursor-not-allowed' : ''
              }`}
            >
              {loginMutation.isPending ? 'Signing in...' : 'Sign in'}
            </button>
          </div>
        </form>
      </div>
    </div>
  );
};

export { Login };
