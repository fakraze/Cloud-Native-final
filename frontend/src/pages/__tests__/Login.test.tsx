import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render, screen, waitFor } from '../../test/test-utils';
import userEvent from '@testing-library/user-event';
import { Login } from '../Login';
import { Navigate } from 'react-router-dom';

// Mock React Router Navigate
vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');
  return {
    ...actual,
    Navigate: vi.fn(() => null),
  };
});

describe('Login Component', () => {
  const mockMutate = vi.fn();

  beforeEach(() => {
    vi.clearAllMocks();
    
    // Reset mocks to default state
    vi.mocked(require('../../hooks/useAuth').useLogin).mockReturnValue({
      mutate: mockMutate,
      isLoading: false,
      error: null
    });

    vi.mocked(require('../../store/authStore').useAuthStore).mockReturnValue({
      isAuthenticated: false,
      user: null
    });
  });

  describe('Rendering', () => {
    it('should render login form when not authenticated', () => {
      render(<Login />);
      
      expect(screen.getByPlaceholderText('Enter your email address')).toBeInTheDocument();
      expect(screen.getByPlaceholderText('Enter your password')).toBeInTheDocument();
      expect(screen.getByRole('button', { name: /sign in/i })).toBeInTheDocument();
    });

    it('should render password toggle button', () => {
      render(<Login />);
      
      const toggleButton = screen.getByLabelText(/toggle password visibility/i);
      expect(toggleButton).toBeInTheDocument();
    });

    it('should render loading state', () => {
      vi.mocked(require('../../hooks/useAuth').useLogin).mockReturnValue({
        mutate: mockMutate,
        isLoading: true,
        error: null
      });

      render(<Login />);
      
      const submitButton = screen.getByRole('button', { name: /signing in/i });
      expect(submitButton).toBeDisabled();
    });

    it('should render error message when login fails', () => {
      vi.mocked(require('../../hooks/useAuth').useLogin).mockReturnValue({
        mutate: mockMutate,
        isLoading: false,
        error: { message: 'Invalid credentials' }
      });

      render(<Login />);
      
      expect(screen.getByText('Invalid credentials')).toBeInTheDocument();
    });
  });

  describe('Form Interactions', () => {
    it('should handle email input changes', async () => {
      const user = userEvent.setup();
      render(<Login />);
      
      const emailInput = screen.getByPlaceholderText('Enter your email address');
      await user.type(emailInput, 'test@example.com');
      
      expect(emailInput).toHaveValue('test@example.com');
    });

    it('should handle password input changes', async () => {
      const user = userEvent.setup();
      render(<Login />);
      
      const passwordInput = screen.getByPlaceholderText('Enter your password');
      await user.type(passwordInput, 'password123');
      
      expect(passwordInput).toHaveValue('password123');
    });

    it('should toggle password visibility', async () => {
      const user = userEvent.setup();
      render(<Login />);
      
      const passwordInput = screen.getByPlaceholderText('Enter your password');
      const toggleButton = screen.getByLabelText(/toggle password visibility/i);
      
      expect(passwordInput).toHaveAttribute('type', 'password');
      
      await user.click(toggleButton);
      expect(passwordInput).toHaveAttribute('type', 'text');
      
      await user.click(toggleButton);
      expect(passwordInput).toHaveAttribute('type', 'password');
    });
  });

  describe('Form Submission', () => {
    it('should submit form with email and password', async () => {
      const user = userEvent.setup();
      render(<Login />);
      
      const emailInput = screen.getByPlaceholderText('Enter your email address');
      const passwordInput = screen.getByPlaceholderText('Enter your password');
      const submitButton = screen.getByRole('button', { name: /sign in/i });
      
      await user.type(emailInput, 'test@example.com');
      await user.type(passwordInput, 'password123');
      await user.click(submitButton);
      
      expect(mockMutate).toHaveBeenCalledWith({
        email: 'test@example.com',
        password: 'password123'
      });
    });

    it('should not submit form when fields are empty', async () => {
      const user = userEvent.setup();
      render(<Login />);
      
      const submitButton = screen.getByRole('button', { name: /sign in/i });
      await user.click(submitButton);
      
      expect(mockMutate).not.toHaveBeenCalled();
    });
  });

  describe('Authentication Redirects', () => {
    it('should redirect admin users to admin dashboard', () => {
      vi.mocked(require('../../store/authStore').useAuthStore).mockReturnValue({
        isAuthenticated: true,
        user: { role: 'admin', id: '1', email: 'admin@test.com' }
      });

      render(<Login />);
      
      expect(Navigate).toHaveBeenCalledWith({ to: '/admin', replace: true }, {});
    });

    it('should redirect staff users to staff dashboard', () => {
      vi.mocked(require('../../store/authStore').useAuthStore).mockReturnValue({
        isAuthenticated: true,
        user: { role: 'staff', id: '2', email: 'staff@test.com' }
      });

      render(<Login />);
      
      expect(Navigate).toHaveBeenCalledWith({ to: '/staff', replace: true }, {});
    });

    it('should redirect customer users to restaurant list', () => {
      vi.mocked(require('../../store/authStore').useAuthStore).mockReturnValue({
        isAuthenticated: true,
        user: { role: 'customer', id: '3', email: 'customer@test.com' }
      });

      render(<Login />);
      
      expect(Navigate).toHaveBeenCalledWith({ to: '/restaurant', replace: true }, {});
    });
  });

  describe('Accessibility', () => {
    it('should have proper form labels and attributes', () => {
      render(<Login />);
      
      const emailInput = screen.getByPlaceholderText('Enter your email address');
      const passwordInput = screen.getByPlaceholderText('Enter your password');
      
      expect(emailInput).toHaveAttribute('type', 'email');
      expect(passwordInput).toHaveAttribute('type', 'password');
      expect(emailInput).toHaveAttribute('required');
      expect(passwordInput).toHaveAttribute('required');
    });
  });

  describe('Snapshots', () => {
    it('should match snapshot in default state', () => {
      const { container } = render(<Login />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot in loading state', () => {
      vi.mocked(require('../../hooks/useAuth').useLogin).mockReturnValue({
        mutate: mockMutate,
        isLoading: true,
        error: null
      });

      const { container } = render(<Login />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot with error', () => {
      vi.mocked(require('../../hooks/useAuth').useLogin).mockReturnValue({
        mutate: mockMutate,
        isLoading: false,
        error: { message: 'Login failed' }
      });

      const { container } = render(<Login />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
