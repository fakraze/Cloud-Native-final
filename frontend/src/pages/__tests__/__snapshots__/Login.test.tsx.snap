// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Login Component > should render login form 1`] = `
<div
  class="min-h-screen bg-gradient-to-br from-slate-50 via-blue-50 to-indigo-100 flex items-center justify-center py-12 px-4 sm:px-6 lg:px-8 relative overflow-hidden"
>
  <div
    class="absolute inset-0 overflow-hidden"
  >
    <div
      class="absolute -top-40 -right-40 w-96 h-96 bg-gradient-to-br from-blue-400/30 to-indigo-600/30 rounded-full blur-3xl animate-pulse"
    />
    <div
      class="absolute -bottom-40 -left-40 w-96 h-96 bg-gradient-to-tr from-purple-400/30 to-pink-600/30 rounded-full blur-3xl animate-pulse animation-delay-1000"
    />
    <div
      class="absolute top-20 left-1/4 w-64 h-64 bg-gradient-to-br from-cyan-400/20 to-blue-500/20 rounded-full blur-3xl animate-pulse animation-delay-2000"
    />
    <div
      class="absolute top-1/4 left-1/3 w-2 h-2 bg-blue-400/60 rounded-full animate-bounce animation-delay-500"
    />
    <div
      class="absolute top-1/3 right-1/4 w-3 h-3 bg-indigo-400/60 rounded-full animate-bounce animation-delay-1500"
    />
    <div
      class="absolute bottom-1/3 left-1/5 w-2 h-2 bg-purple-400/60 rounded-full animate-bounce animation-delay-3000"
    />
    <div
      class="absolute bottom-1/4 right-1/3 w-1 h-1 bg-pink-400/60 rounded-full animate-bounce animation-delay-2500"
    />
    <div
      class="absolute top-10 right-10 w-12 h-12 border-2 border-blue-300/30 rotate-45 animate-spin"
      style="animation-duration: 20s;"
    />
    <div
      class="absolute bottom-10 left-10 w-8 h-8 border-2 border-purple-300/30 rotate-12 animate-spin"
      style="animation-duration: 15s;"
    />
  </div>
        
  <div
    class="max-w-md w-full space-y-8 relative z-10"
  >
    <div
      class="text-center"
    >
      <div
        class="mx-auto h-20 w-20 bg-gradient-to-br from-blue-600 via-indigo-600 to-purple-600 rounded-3xl flex items-center justify-center shadow-2xl mb-8 transform hover:scale-110 transition-all duration-500 hover:rotate-3 relative overflow-hidden group"
      >
        <div
          class="absolute inset-0 bg-gradient-to-br from-white/20 to-transparent opacity-0 group-hover:opacity-100 transition-opacity duration-300"
        />
        <svg
          aria-hidden="true"
          class="lucide lucide-mail h-10 w-10 text-white relative z-10 group-hover:scale-110 transition-transform duration-300"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m22 7-8.991 5.727a2 2 0 0 1-2.009 0L2 7"
          />
          <rect
            height="16"
            rx="2"
            width="20"
            x="2"
            y="4"
          />
        </svg>
        <div
          class="absolute inset-0 rounded-3xl bg-gradient-to-br from-blue-400/50 to-transparent animate-pulse"
        />
      </div>
      <h2
        class="text-5xl font-extrabold bg-gradient-to-r from-gray-900 via-blue-800 to-indigo-900 bg-clip-text text-transparent mb-4 tracking-tight"
      >
        Welcome Back
      </h2>
      <div
        class="h-1 w-24 bg-gradient-to-r from-blue-500 to-indigo-500 mx-auto rounded-full mb-4"
      />
      <p
        class="text-xl text-gray-600 font-medium mb-2"
      >
        Restaurant Ordering System
      </p>
      <p
        class="text-sm text-gray-500 italic"
      >
        Sign in to continue your culinary journey
      </p>
                
    </div>
            
    <form
      class="glass-card p-8 space-y-6 hover:shadow-2xl transition-all duration-500 transform hover:-translate-y-1"
    >
      <div
        class="space-y-6"
      >
        <div
          class="group"
        >
          <label
            class="block text-sm font-bold text-gray-700 mb-3 group-focus-within:text-blue-600 transition-colors duration-200"
            for="email"
          >
            Email address
          </label>
          <div
            class="relative group"
          >
            <div
              class="absolute inset-y-0 left-0 pl-4 flex items-center pointer-events-none"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-mail h-5 w-5 text-gray-400 group-focus-within:text-blue-500 transition-all duration-300 group-focus-within:scale-110"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m22 7-8.991 5.727a2 2 0 0 1-2.009 0L2 7"
                />
                <rect
                  height="16"
                  rx="2"
                  width="20"
                  x="2"
                  y="4"
                />
              </svg>
            </div>
            <input
              autocomplete="email"
              class="input-field pl-12 text-gray-900 placeholder-gray-500 hover:border-blue-300 focus:scale-105 transition-all duration-300"
              id="email"
              name="email"
              placeholder="Enter your email address"
              required=""
              type="email"
              value=""
            />
          </div>
        </div>
        <div
          class="group"
        >
          <label
            class="block text-sm font-bold text-gray-700 mb-3 group-focus-within:text-blue-600 transition-colors duration-200"
            for="password"
          >
            Password
          </label>
          <div
            class="relative group"
          >
            <div
              class="absolute inset-y-0 left-0 pl-4 flex items-center pointer-events-none"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-lock h-5 w-5 text-gray-400 group-focus-within:text-blue-500 transition-all duration-300 group-focus-within:scale-110"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="11"
                  rx="2"
                  ry="2"
                  width="18"
                  x="3"
                  y="11"
                />
                <path
                  d="M7 11V7a5 5 0 0 1 10 0v4"
                />
              </svg>
            </div>
            <input
              autocomplete="current-password"
              class="input-field pl-12 pr-12 text-gray-900 placeholder-gray-500 hover:border-blue-300 focus:scale-105 transition-all duration-300"
              id="password"
              name="password"
              placeholder="Enter your password"
              required=""
              type="password"
              value=""
            />
            <button
              class="absolute inset-y-0 right-0 pr-4 flex items-center hover:bg-gray-50 rounded-r-xl transition-all duration-200 hover:scale-110"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-eye h-5 w-5 text-gray-400 hover:text-gray-600 transition-colors duration-200"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
                
      <div
        class="pt-4"
      >
        <button
          class="w-full btn-primary relative overflow-hidden group hover:shadow-2xl hover:scale-105"
          type="submit"
        >
          <div
            class="absolute inset-0 bg-gradient-to-r from-transparent via-white/20 to-transparent -translate-x-full group-hover:translate-x-full transition-transform duration-1000"
          />
          <span
            class="relative z-10 flex items-center justify-center font-bold text-lg "
          >
            Sign in
            <span
              class="ml-2 group-hover:translate-x-1 transition-transform duration-200"
            >
              →
            </span>
          </span>
                      
        </button>
      </div>
      <div
        class="text-center pt-4"
      >
        <p
          class="text-sm text-gray-500"
        >
          Secure login powered by
           
          <span
            class="bg-gradient-to-r from-blue-600 to-indigo-600 bg-clip-text text-transparent font-semibold"
          >
            Cloud Native Architecture
          </span>
        </p>
      </div>
    </form>
    <div
      class="text-center mt-8"
    >
      <div
        class="flex items-center justify-center space-x-2 text-gray-400 text-sm"
      >
        <div
          class="w-8 h-px bg-gradient-to-r from-transparent to-gray-300"
        />
        <span>
          Crafted with ❤️ for food lovers
        </span>
        <div
          class="w-8 h-px bg-gradient-to-l from-transparent to-gray-300"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Login Component > should render login form with proper styling and structure 1`] = `
<form
  class="glass-card p-8 space-y-6 hover:shadow-2xl transition-all duration-500 transform hover:-translate-y-1"
>
  <div
    class="space-y-6"
  >
    <div
      class="group"
    >
      <label
        class="block text-sm font-bold text-gray-700 mb-3 group-focus-within:text-blue-600 transition-colors duration-200"
        for="email"
      >
        Email address
      </label>
      <div
        class="relative group"
      >
        <div
          class="absolute inset-y-0 left-0 pl-4 flex items-center pointer-events-none"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-mail h-5 w-5 text-gray-400 group-focus-within:text-blue-500 transition-all duration-300 group-focus-within:scale-110"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m22 7-8.991 5.727a2 2 0 0 1-2.009 0L2 7"
            />
            <rect
              height="16"
              rx="2"
              width="20"
              x="2"
              y="4"
            />
          </svg>
        </div>
        <input
          autocomplete="email"
          class="input-field pl-12 text-gray-900 placeholder-gray-500 hover:border-blue-300 focus:scale-105 transition-all duration-300"
          id="email"
          name="email"
          placeholder="Enter your email address"
          required=""
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="group"
    >
      <label
        class="block text-sm font-bold text-gray-700 mb-3 group-focus-within:text-blue-600 transition-colors duration-200"
        for="password"
      >
        Password
      </label>
      <div
        class="relative group"
      >
        <div
          class="absolute inset-y-0 left-0 pl-4 flex items-center pointer-events-none"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-lock h-5 w-5 text-gray-400 group-focus-within:text-blue-500 transition-all duration-300 group-focus-within:scale-110"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="11"
              rx="2"
              ry="2"
              width="18"
              x="3"
              y="11"
            />
            <path
              d="M7 11V7a5 5 0 0 1 10 0v4"
            />
          </svg>
        </div>
        <input
          autocomplete="current-password"
          class="input-field pl-12 pr-12 text-gray-900 placeholder-gray-500 hover:border-blue-300 focus:scale-105 transition-all duration-300"
          id="password"
          name="password"
          placeholder="Enter your password"
          required=""
          type="password"
          value=""
        />
        <button
          class="absolute inset-y-0 right-0 pr-4 flex items-center hover:bg-gray-50 rounded-r-xl transition-all duration-200 hover:scale-110"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-eye h-5 w-5 text-gray-400 hover:text-gray-600 transition-colors duration-200"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
            />
            <circle
              cx="12"
              cy="12"
              r="3"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
            
  <div
    class="pt-4"
  >
    <button
      class="w-full btn-primary relative overflow-hidden group hover:shadow-2xl hover:scale-105"
      type="submit"
    >
      <div
        class="absolute inset-0 bg-gradient-to-r from-transparent via-white/20 to-transparent -translate-x-full group-hover:translate-x-full transition-transform duration-1000"
      />
      <span
        class="relative z-10 flex items-center justify-center font-bold text-lg "
      >
        Sign in
        <span
          class="ml-2 group-hover:translate-x-1 transition-transform duration-200"
        >
          →
        </span>
      </span>
                  
    </button>
  </div>
  <div
    class="text-center pt-4"
  >
    <p
      class="text-sm text-gray-500"
    >
      Secure login powered by
       
      <span
        class="bg-gradient-to-r from-blue-600 to-indigo-600 bg-clip-text text-transparent font-semibold"
      >
        Cloud Native Architecture
      </span>
    </p>
  </div>
</form>
`;

exports[`Login Component > should render login form with proper styling and structure 2`] = `
<input
  autocomplete="email"
  class="input-field pl-12 text-gray-900 placeholder-gray-500 hover:border-blue-300 focus:scale-105 transition-all duration-300"
  id="email"
  name="email"
  placeholder="Enter your email address"
  required=""
  type="email"
  value=""
/>
`;

exports[`Login Component > should render login form with proper styling and structure 3`] = `
<input
  autocomplete="current-password"
  class="input-field pl-12 pr-12 text-gray-900 placeholder-gray-500 hover:border-blue-300 focus:scale-105 transition-all duration-300"
  id="password"
  name="password"
  placeholder="Enter your password"
  required=""
  type="password"
  value=""
/>
`;

exports[`Login Component > should render submit button 1`] = `
<button
  class="w-full btn-primary relative overflow-hidden group hover:shadow-2xl hover:scale-105"
  type="submit"
>
  <div
    class="absolute inset-0 bg-gradient-to-r from-transparent via-white/20 to-transparent -translate-x-full group-hover:translate-x-full transition-transform duration-1000"
  />
  <span
    class="relative z-10 flex items-center justify-center font-bold text-lg "
  >
    Sign in
    <span
      class="ml-2 group-hover:translate-x-1 transition-transform duration-200"
    >
      →
    </span>
  </span>
              
</button>
`;
